pkg/scheduler/redis_scheduler.go

```diff
@@ -22,91 +22,126 @@ const TriggersKey = "triggers"
 // currently being executed
 const SchedKey = "scheduling"
 
-// luaPoll returns the lua script used for polling triggers in redis
-// TODO we should poll sched too
-const luaPoll = `
-local res = redis.call("ZRANGEBYSCORE", "` + TriggersKey + `", 0, KEYS[1], "WITHSCORES", "LIMIT", 0, 1)
-if #res > 0 then
-  redis.call("ZREM", "` + TriggersKey + `", res[1])
-  redis.call("ZADD", "` + SchedKey + `", res[2], res[1])
-end
-return res`
-
 // pollInterval is the time interval between 2 redis polling
 const pollInterval = 1 * time.Second
 
+// luaPoll returns the lua script used for polling triggers in redis.
+// If a trigger is in the scheduling key for more than 10 seconds, it is
+// an error and we can try again to schedule it.
+const luaPoll = `
+local w = KEYS[1] - 10
+local s = redis.call("ZRANGEBYSCORE", "` + SchedKey + `", 0, w, "WITHSCORES", "LIMIT", 0, 1)
+if #s > 0 then
+  redis.call("ZADD", "` + SchedKey + `", KEYS[1], s[1])
+  return s
+end
+local t = redis.call("ZRANGEBYSCORE", "` + TriggersKey + `", 0, KEYS[1], "WITHSCORES", "LIMIT", 0, 1)
+if #t > 0 then
+  redis.call("ZREM", "` + TriggersKey + `", t[1])
+  redis.call("ZADD", "` + SchedKey + `", t[2], t[1])
+end
+return t`
+
 // RedisScheduler is a centralized scheduler of many triggers. It starts all of
 // them and schedules jobs accordingly.
 type RedisScheduler struct {
-	Broker jobs.Broker
-	client *redis.Client
+	broker  jobs.Broker
+	client  *redis.Client
+	stopped chan struct{}
 }
 
 // NewRedisScheduler creates a new scheduler that use redis to synchronize with
 // other cozy-stack processes to schedule jobs.
 func NewRedisScheduler(client *redis.Client) *RedisScheduler {
 	return &RedisScheduler{
-		client: client,
+		client:  client,
+		stopped: make(chan struct{}),
 	}
 }
 
 func redisKey(infos *TriggerInfos) string {
-	return infos.Domain + ":" + infos.TID
+	return infos.Domain + "/" + infos.TID
 }
 
 // Start a goroutine that will fetch triggers in redis to schedule their jobs
 func (s *RedisScheduler) Start(b jobs.Broker) error {
-	s.Broker = b
+	s.broker = b
 	go func() {
-		for range time.Tick(pollInterval) {
-			if err := s.Poll(); err != nil {
-				log.Warnf("[Scheduler] Failed to poll redis: %s", err)
+		ticker := time.NewTicker(pollInterval)
```

Comment:
you could introduce and random delay before starting this ticker to avoid synchronous polling between stacks.